tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/CrossTestEncodings.scala

```diff
@@ -335,15 +341,54 @@ trait CrossTestEncodings extends AnyFunSuite with Checkers {
     }
   }
 
-  // Ignore this test until the bug described in https://github.com/informalsystems/apalache/issues/1660 is fixed.
+  @nowarn("cat=deprecation&msg=Stream") // ScalaCheck's Shrink still uses Stream, which is deprecated since Scala 2.13.0
+  private def shrinkType(tp: TlaType1): Stream[TlaType1] = tp match {
+    case IntT1() | RealT1() | BoolT1() | StrT1() => Stream.empty
+    case FunT1(arg, res)                         => Stream(arg, res)
+    case SetT1(elem)                             => Stream(elem)
+    case SeqT1(elem)                             => Stream(elem)
+    case TupT1(elems @ _*) =>
+      Stream.concat(
+          // shrink to one of the element types
+          Stream(elems: _*),
+          // recursively shrink one tuple element type
+          elems.zipWithIndex.foldLeft(Stream.empty[TlaType1]) { case (acc, (tp, idx)) =>
+            acc ++ shrink(tp).map(shrunkType => TupT1(elems.updated(idx, shrunkType): _*))
+          },
+          // tuples over a subset of the current element types
+          Stream(shrink(elems).filterNot(_.isEmpty).map(elems => TupT1(elems: _*)): _*),
+      )
+    case RecT1(fieldTypes) =>
+      Stream.concat(
+          // shrink to one of the field types
+          Stream(fieldTypes.values.toSeq: _*),
+          // recursively shrink one record field type
+          fieldTypes.foldLeft(Stream.empty[TlaType1]) { case (acc, (fieldName, tp)) =>
+            acc ++ shrink(tp).map(shrunkType => RecT1(fieldTypes.updated(fieldName, shrunkType)))
+          },
+          // tuples over a subset of the current field types
+          Stream(shrink(fieldTypes).filterNot(_.isEmpty).map(elems => RecT1(elems)): _*),
+      )
+    case ConstT1(_) | OperT1(_, _) | RealT1() | RecRowT1(_) | RowT1(_, _) | SparseTupT1(_) | VarT1(_) | VariantT1(_) =>
+      throw new IllegalArgumentException(s"Shrinking unsupported type ${tp}. Should be disabled in type generator.")
```

Comment:
Do I understand correctly that this is because we don't generate expressions of these types? If so, I think I'd find an error like this a bit clearer:

```suggestion
    case ConstT1(_) | OperT1(_, _) | RealT1() | RecRowT1(_) | RowT1(_, _) | SparseTupT1(_) | VarT1(_) | VariantT1(_) =>
      throw new IllegalArgumentException(s"Shrinking type ${tp} that is is not currently supported by the type generator.")
```

I think the added clarity from my POV is that the other could be read as saying that the type is unsupported per se, and that the fix is to disable it in the type generator. But, iiuc, it's because we presume we've disabled these from the type generator that it is unsupported.

Reply:
Not entirely accurate, `TlaType1Gen` is generating them, I'm ad-hoc overriding the generator at the top of the file, see the declaration of member `typeGen`.

I've modified your suggestion to account for that.